hooks/checkout

```diff
@@ -84,6 +84,11 @@ checkout() {
   git reset --hard
   git clean -ffxdq
 
+  # Check the current state to make sure we start from a clean working tree.
+  # This does both "refresh the index and updating the cached stat information"
+  # as per `man git-status` **BACKGROUND REFRESH**.
+  git status
```

Comment:
The `status` should normally display a clean state so it shouldn't add to much noise to the logs